larry/backup.go

```diff
@@ -0,0 +1,275 @@
+// Copyright (c) 2025 Hemi Labs, Inc.
+// Use of this source code is governed by the MIT License,
+// which can be found in the LICENSE file.
+
+package larry
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/dustin/go-humanize"
+)
+
+// DumpTables and Restore are used to create portable backups. Any
+// Encoder/Decoder can be used provided it conforms to the interface.  The
+// caller is responsible for quiescing the tables. The backup itself will be
+// made using a consistent view.
+//
+// Typical use:
+//	var b bytes.Buffer
+//	zw, _ := zstd.NewWriter(&b)
+//	je := json.NewEncoder(zw)
+//	_ = db.DumpTables(ctx, tables, je)
+//	_ = zw.Close()
+//
+//	zr, _ := zstd.NewReader(&b)
+//	jd := json.NewDecoder(zr)
+//	_ = db.Restore(ctx, jd)
+
+// OperationT is used to tag what database operation is being described. These
+// operations are used to perform database dump/restores.
+type OperationT uint8
+
+const (
+	OpPut OperationT = 1 // Database put
+	OpDel OperationT = 2 // Database delete
+)
+
+// Operation describes a single database operation. An entire database can be
+// replicated by creating a delta between a source database table and a
+// destination database table.
+type Operation struct {
+	Op    OperationT `json:"op"`
+	Table string     `json:"table"`
+	Key   []byte     `json:"key"`
+	Value []byte     `json:"value,omitempty" binary:"omitempty"`
+}
+
+// Replay describes a replay of operations into a database.
+type Replay struct {
+	Operations []Operation `json:"operations"`
+}
+
+type Encoder interface {
+	Encode(v any) error
+}
+
+type Decoder interface {
+	Decode(v any) error
+}
+
+var (
+	DefaultMaxRestoreChunk = 256 * 1024 * 1024 // Maximum transaction size during restore.
+
+	Verbose = true // use logger to print progress
```

Comment:
A few questions:
1. Verbose seems to be used as a gate to use Info logs. Would it potentially be better to use debug/trace logs instead of having a separate version flag? With loggo, the logger level can be set per-module (name provided when creating a logger).
2. If we do want to keep this flag, should it default to `false` instead?